index.js

```diff
@@ -280,6 +281,84 @@ app.get("/label", async (req, res) => {
 
 //#endregion
 
+//#region  --- /user ROUTES
+
+// Register a new user
+app.post('/register', async (req, res) => {
+    try {
+      const {firstname, lastname, username, email, password } = req.body;
+
+      // Hash the password
+      const hashedPassword = await bcrypt.hash(password, 10);
+
+      // Check if user already exists
+      const user = await getUsersByUsername(username);
+      if (user.length !== 0) {
+        return res.status(401).send('Username or email already exists');
+      }
+
+      // Insert the new user into the database
+      const newAccount = await newUser(firstname, lastname, username, email, hashedPassword);
+      if (newAccount.length === 0){
+        res.status(401).json({ error: 'Failed to create account' });
```

Comment:
401 implies the user is unauthorized to perform the action when it's more of an internal error. Maybe a 500 code? I'm not too sure and I need to review what codes I'm using in the code I worked on too because I don't think it's exactly right.